python/ray/_private/utils.py

```diff
@@ -322,6 +342,129 @@ def get_aws_neuron_core_visible_ids() -> Optional[List[str]]:
     return _get_visible_ids(env_var=ray_constants.NEURON_RT_VISIBLE_CORES_ENV_VAR)
 
 
+def has_intel_gpu() -> bool:
+    """
+    Returns:
+        Return True if has visible Intel GPU
+    """
+    gpu_ids = get_intel_gpu_visible_ids()
+    if gpu_ids is not None and len(gpu_ids) > 0:
+        return True
+    return False
+
+
+def has_nvidia_gpu() -> bool:
+    """
+    Returns:
+        Return True if has visible Nvidia GPU
+    """
+    gpu_ids = get_nvidia_gpu_visible_ids()
+    if gpu_ids is not None and len(gpu_ids) > 0:
+        return True
+    return False
+
+
+def get_intel_gpu_visible_ids() -> Optional[List[str]]:
+    """
+    Get the device IDs using ONEAPI_DEVICE_SELECTOR environment variable.
+
+    Return:
+        devices (List[str]): If environment variable is set, returns a
+            list of strings representing the relative IDS of the visible devices.
+            If it is not set, returns empty None.
+    """
+    if os.environ.get("ONEAPI_DEVICE_SELECTOR", None) is None:
+        return None
```

Comment:
This seems to be conflict with the meaning of the environment variable "With no environment variables set to say otherwise, all platforms and devices presently on the machine are available"?